### PR TITLE
Task 6 (SQS & SNS, Async Microservices Communication)

### PR DESCRIPTION
**Pull request by tuwhoo**
- [x] File `serverless.yml` contains configuration for `catalogBatchProcess` function
- [x] File `serverless.yml` contains policies to allow lambda `catalogBatchProcess` function to interact with `SNS` and `SQS`
- [x] File `serverless.yml` contains configuration for SQS `catalogItemsQueue`
- [x] File `serverless.yml` contains configuration for SNS Topic `createProductTopic` and email subscription

### CODE REVIEW

File: product-service/serverless.ts [hunk 1]
```diff
@@ -44,11 +46,36 @@ const serverlessConfiguration: AWS = {
             ],
             Resource: ['${env:PRODUCTS_TABLE_ARN}', '${env:STOCKS_TABLE_ARN}']
           },
+          {
+            Effect: "Allow",
+            Action: ["sns:*"],
+            Resource: {
+              Ref: "SNSTopic",
+            },
+          },
```
Comment (ppparfenovich) [inline product-service/serverless.ts:55]: I think here should be permission for SQS actions

File: product-service/serverless.ts [hunk 2]
```diff
@@ -44,11 +46,36 @@ const serverlessConfiguration: AWS = {
...
+      SNSSubscription: {
+        Type: "AWS::SNS::Subscription",
+        Properties: {
+          Endpoint: "mikita_trukhanavets@epam.com",
+          Protocol: "email",
+          TopicArn: {
+            Ref: "SNSTopic",
+          },
+        },
+      },
```
Comment (ppparfenovich) [inline product-service/serverless.ts:75]: Same for resources. Should be provided resources for SQS.

Comment (tuwhoo) [reply product-service/serverless.ts:75]: Thanks for your comment. I configured SQS Service in `import-service/serverless.ts` and share it with `product-service` via arn in `.env` file. I'm not sure it's very important in which exactly service it should be declared, cause it is used in both services.

File: product-service/serverless.ts (hunk 1 above)
Comment (tuwhoo) [reply product-service/serverless.ts:55]: look `product-service/src/functions/catalogBatchProcess/index.ts`

**Comment by rostislav-sergeenkov** [comment]
Reviewed. Total Score: 4

**Pull request merged**